contracts/test/LightClientBenchmark.t.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: Unlicensed
+
+/* solhint-disable contract-name-camelcase, func-name-mixedcase, one-contract-per-file */
+
+pragma solidity ^0.8.0;
+
+// Libraries
+import "forge-std/Test.sol";
+import { IPlonkVerifier as V } from "../src/interfaces/IPlonkVerifier.sol";
+
+// Target contract
+import { LightClient as LC } from "../src/LightClient.sol";
+import { LightClientCommonTest } from "./LightClient.t.sol";
+
+contract LightClient_newFinalizedState_Test is LightClientCommonTest {
+    function setUp() public {
+        init();
+    }
+
+    /// @dev for benchmarking purposes only
+    function testCorrectUpdateBench() external {
+        // Generating a few consecutive states and proofs
+        string[] memory cmds = new string[](6);
+        cmds[0] = "diff-test";
+        cmds[1] = "mock-consecutive-finalized-states";
+        cmds[2] = vm.toString(BLOCKS_PER_EPOCH_TEST);
+        cmds[3] = vm.toString(STAKE_TABLE_CAPACITY / 2);
+        cmds[4] = vm.toString(uint64(3));
+        cmds[5] = vm.toString(uint64(3));
+
+        bytes memory result = vm.ffi(cmds);
+        (LC.LightClientState[] memory states, V.PlonkProof[] memory proofs) =
+            abi.decode(result, (LC.LightClientState[], V.PlonkProof[]));
+        vm.prank(permissionedProver);
+        lc.newFinalizedState(states[0], proofs[0]);
+    }
```

Comment:
to get a focused/accurate overall gas consumption, maybe add [`vm.pauseGasMetering()`](https://book.getfoundry.sh/cheatcodes/pause-gas-metering) in the first line of `setup()` and add [`vm.resumeGasMetering()`](https://book.getfoundry.sh/cheatcodes/resume-gas-metering) before calling `lc.newFinalizedState()`

Reply:
Nice! We are now below 600k!
`[PASS] testCorrectUpdateBench() (gas: 597104)`

Change done in e903e6b0.